src/main/java/hudson/plugins/git/BranchSpec.java

```diff
@@ -97,7 +97,11 @@ public boolean matches(String item, EnvVars env) {
     }
 
     private String getExpandedName(EnvVars env) {
-        return env.expand(name);
+        String expandedName = env.expand(name);
+        if (expandedName.length() == 0) {
+            return "**";
```

Comment:
I really should have added comments, this one is a pain to remember.

I think I added this because otherwise there was a behavior discrepancy between an empty branchspec, and a branchpec that resolves to empty. I can't find it now, there should be a short path somewhere.
Or maybe it's a leftover from my first attempts. I could test without it, but I'll have to wait until tonight.